Add Tags to blog

Cahllagerfeld opened the pull request:
closes #3021

For better UX, SSR gets enabled for the `/blog` route. otherwise the filtering would happen `onMount` which looks a bit weird to the user

Added a new prompt to the `npm run generate:blog-post` when creating a new post

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3239"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

Siddhant-K-code commented on src/app.d.ts:
```diff
@@ -18,6 +18,7 @@ declare global {
       guides: any[];
       securityLogs: any[];
       educationCustomers: any[];
+      filter: string;
```
**(early) suggestion:** can we use something as `tag` or `category`? wdyt?

URL Slug would be something like => `/blog?tag=Engineering`

Cahllagerfeld reviewed (commented):
Hey @chrifro I added the tags from notion to the corresponding blog posts. There are still a lot of posts missing without one. Do we need to add them within this PR, or shall we do this as a follow up if required?

chrifro commented:
Hi @Cahllagerfeld , the ones mentioned in the RFC are just some examples. Can you add the tags to all other blogs post? If you are uncertain about any, you could make a table overview for them and we figure it out together. I'd say all blog posts since 2021 should have a tag.

chrifro reviewed (commented):
Awesome @Cahllagerfeld  ✨


### **Some comments:**
**on/blog**

- [x]  the hover for the filters is missing

<img width="611" alt="Bildschirm­foto 2023-01-16 um 09 17 17" src="https://user-images.githubusercontent.com/46345125/212629776-e677d30f-217c-4b20-a075-a7fa7d50540b.png">

- [x]  I used the [secondary button style](https://www.figma.com/file/VL0CJpHqcDJVJ4BEkoyRFV/Gitpod-Website-Component-Library?node-id=513%3A527&t=tWnkDc3sL5udmUrq-1) which has a defined height of 32 px. Your button is 37px in height. What's your intention behind that?

- [ ] on dark theme: the label shadow is off. Can also be addressed in #1950 [see [component library](https://www.figma.com/file/VL0CJpHqcDJVJ4BEkoyRFV/Gitpod-Website-Component-Library?node-id=557%3A1088&t=tWnkDc3sL5udmUrq-1)]

<img width="285" alt="Bildschirm­foto 2023-01-16 um 09 24 38" src="https://user-images.githubusercontent.com/46345125/212631544-9b3aaf21-b040-45d4-b8d3-e0c714bec85e.png">
 
 **On /blog/xyz**:

- [ ] when being on a blog post about "engineering" the "you might also like section" should only display other blog posts with the same label. Let's display 3 random blog posts with the same label from the last 12 months. 
 
 
 - how can we make sure that a new tag will be added for new blog posts? Any ideas? 
 
unrelated: 
 - the minimum spacing between text and authors should be 32 px
<img width="355" alt="Bildschirm­foto 2023-01-16 um 09 30 14" src="https://user-images.githubusercontent.com/46345125/212632491-8fdba5d2-4869-4eaa-95cf-4685712a0381.png">

Cahllagerfeld commented:
I'm a bit on the fence with picking random blogposts to display because the result won't be predictable. We may link three different ones after rebuilding the page, which would lead to inconsistencies also with SEO, because the linked pages now link to other pages.

I don't know if this is a bad impact or not though.
Ideally we would have a way to have 3 different posts per post, but in a predictable way I think 🤔

Cahllagerfeld commented:
- added a prng for being able to predict the result of the random generate for the blogposts displayed in the section
- updated the button to use the actual button component
- updated spacing for avatars

chrifro reviewed (approved):
Works great! Awesome solution for displaying the same random blog posts ✨

Let's merge it 🚀
